### PR TITLE
[Python] Added BinarySubscriber and BinaryPublisher with examples

### PR DESCRIPTION
**Pull request by LeonHosch**
### Description
I added a BinarySubscriber and a BinaryPublisher class to the Python wheel. Now you can send binary data using Python

### Related issues

-

### CODE REVIEW

File: samples/python/pubsub/binary/binary_rec/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ========================= eCAL LICENSE =================================
+
+project(binary_rec)
+
+find_package(eCAL REQUIRED)
+
+set(PROJECT_GROUP minimal)
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_rec/CMakeLists.txt:23]: PROJECT_GROUP should be "binary" here

File: samples/python/pubsub/binary/binary_rec/binary_rec.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ecal_core.set_process_state(1, 1, "I feel good")
+
+  # create subscriber
+  sub = BinarySubscriber("Hello")
+  
+  # receive messages
+  while ecal_core.ok():
+    ret, msg, time = sub.receive(500)
+    if ret > 0:
+      print("Received:  {} ms   {}".format(time, msg))
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_rec/binary_rec.py:41]: Embedding the binary into a regular doesn't look very pretty as non-printable characters will be escaped with leading backslashes which make it looks kind of messy . Maybe a hexadecimal representation would be better here such as "FF EE CC 1A".

File: samples/python/pubsub/binary/binary_rec_cb/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ========================= eCAL LICENSE =================================
+
+project(binary_rec_cb)
+
+find_package(eCAL REQUIRED)
+
+set(PROJECT_GROUP minimal)
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_rec_cb/CMakeLists.txt:23]: PROJECT_GROUP is wrong

File: samples/python/pubsub/binary/binary_rec_cb/binary_rec_cb.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import sys
+import time
+
+import ecal.core.core as ecal_core
+from ecal.core.subscriber import BinarySubscriber
+
+# eCAL receive callback
+def callback(topic_name, msg, time):
+  print("Received:  {} ms   {}".format(time, msg))
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_rec_cb/binary_rec_cb.py:27]: Same here with embedded binary data.

File: samples/python/pubsub/binary/binary_snd/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ========================= eCAL LICENSE =================================
+
+project(binary_snd)
+
+find_package(eCAL REQUIRED)
+
+set(PROJECT_GROUP minimal)
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_snd/CMakeLists.txt:23]: wrong PROJECT_GROUP

File: samples/python/pubsub/binary/binary_snd/binary_snd.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  ecal_core.set_process_state(1, 1, "I feel good")
+
+  # create publisher
+  pub = BinaryPublisher("Hello")
+  msg = b"Hello from eCAL"
+  
+  # send messages
+  staticmsg = msg
+  while ecal_core.ok():
+    pub.send(msg)
```
Comment (hannemn) [inline samples/python/pubsub/binary/binary_snd/binary_snd.py:43]: The first send iteration should already contain the random binary array.

**Review by hannemn — CHANGES_REQUESTED** [review]
See comments, I attached to the source files.

**Review by FlorianReimold — APPROVED** [review]

**Pull request merged**